src/components/standalone/firewall/CreateOrEditPortForwardDrawer.vue

```diff
@@ -0,0 +1,389 @@
+const restrictIPValidationErrors = ref<string[]>([])
+
+// Options
+const supportedProtocols = ref<NeComboboxOption[]>([])
+const supportedReflectionZones = ref<NeComboboxOption[]>([])
+const wanInterfaces = ref<NeComboboxOption[]>([])
+const supportedDestinationZones = computed<NeComboboxOption[]>(() => {
+  if (firewallConfig.loading) return []
+  else {
+    return [
+      { id: 'any', label: t('standalone.port_forward.any_zone') },
+      ...firewallConfig.zones
+        .filter((zone) => zone.type() != ZoneType.WAN)
+        .map((zone) => ({
+          id: zone.name,
+          label: zone.name.toUpperCase()
+        }))
+    ]
+  }
+})
+
+// Form fields
+const id = ref('')
+const name = ref('')
+const sourcePort = ref('')
+const destinationIP = ref('')
+const destinationPort = ref('')
+const wan = ref('')
+const enabled = ref(false)
+const restrict = ref<string[]>([])
+const protocols = ref<NeComboboxOption[]>([])
+const log = ref(false)
+const reflection = ref(false)
+const reflectionZones = ref<NeComboboxOption[]>([])
+const destinationZone = ref('')
+
+function resetForm() {
+  validationErrorBag.value.clear()
+
+  id.value = props.initialItem?.id ?? ''
+  name.value = props.initialItem?.name ?? ''
+  sourcePort.value = props.initialItem?.source_port ?? ''
+  destinationIP.value = props.initialItem?.dest_ip ?? ''
+  destinationPort.value = props.initialItem?.destination_port ?? ''
+  wan.value = props.initialItem?.wan ?? 'any'
+  enabled.value = props.initialItem?.enabled ?? false
+  restrict.value = props.initialItem?.restrict.map((x) => x) ?? []
+  protocols.value =
+    props.initialItem?.protocol.map((proto: string) => ({
+      id: proto,
+      label: proto.toUpperCase()
+    })) ?? []
+  log.value = props.initialItem?.log ?? false
+  reflection.value = props.initialItem?.reflection ?? false
+  reflectionZones.value =
+    props.initialItem?.reflection_zone.map((reflectionZone) => ({
+      id: reflectionZone,
+      label: reflectionZone.toUpperCase()
+    })) ?? []
+  destinationZone.value = props.initialItem?.dest ?? 'any'
+}
+
+const { t } = useI18n()
+
+async function fetchOptions() {
+  try {
+    loading.value = true
+    supportedProtocols.value = (
+      await ubusCall('ns.redirects', 'list-protocols')
+    ).data.protocols.map((proto: string) => ({
+      id: proto,
+      label: proto.toUpperCase()
+    }))
+    supportedReflectionZones.value = (await ubusCall('ns.redirects', 'list-zones')).data.zones.map(
+      (zone: string) => ({
+        id: zone,
+        label: zone.toUpperCase()
+      })
+    )
+    wanInterfaces.value = [
+      { id: 'any', label: t('standalone.port_forward.any') },
+      ...(await ubusCall('ns.redirects', 'list-wans')).data.wans.map(
+        (iface: { device: string; ipaddr: string }) => ({
+          id: iface.ipaddr,
+          label: iface.ipaddr,
+          description: iface.device
+        })
+      )
+    ]
+    loading.value = false
+  } catch (err: any) {
```

Comment:
Let's use a different `try/catch` for every API call, in order to display a specific error message ("Cannot retrieve protocols", "Cannot retrieve zones", "Cannot retrieve WAN interfaces")